apps/worker/src/app/workflow/usecases/send-message/send-message-chat.usecase.ts

```diff
@@ -465,112 +577,348 @@ export class SendMessageChat extends SendMessageBase {
+    return {
+      ...(command.overrides[integration?.channel] || {}),
+      ...(command.overrides[integration?.providerId] || {}),
+    };
+  }
+
+  private applyAddressSpecificOverrides<T extends ChannelData>(
+    originalChannelData: T,
+    combinedOverrides: Record<string, unknown>
+  ): T {
+    const { identifier, type } = originalChannelData;
+
+    // Early returns for invalid cases
+    if (!identifier) return originalChannelData;
+
+    const addressOverrides = combinedOverrides[identifier];
+    if (!addressOverrides || typeof addressOverrides !== 'object') {
+      return originalChannelData;
+    }
+
+    const newAddress = (addressOverrides as Record<string, unknown>).address;
+    if (!newAddress || typeof newAddress !== 'object') {
+      return originalChannelData;
+    }
+
+    // Validate the new address against the channel type schema
+    try {
+      validateAddressForType(type, newAddress as Record<string, unknown>);
 
       return {
-        status: SendMessageStatus.FAILED,
-        errorMessage: DetailEnum.PROVIDER_ERROR,
+        ...originalChannelData,
+        address: newAddress as T['address'],
       };
+    } catch (_error) {
+      // ignoring the override since it's invalid
+      return originalChannelData;
     }
   }
+
+  private async handleMessageSendSuccess(
+    result: ISendMessageSuccessResponse,
+    message: MessageEntity,
+    command: SendMessageChannelCommand,
+    channelData: ChannelData
+  ): Promise<SendMessageResult> {
+    const redactedChannelData = this.redactChannelData(channelData);
+
+    await this.createExecutionDetail(
+      command,
+      DetailEnum.MESSAGE_SENT,
+      ExecutionDetailsStatusEnum.SUCCESS,
+      message._id,
+      {
+        ...result,
+        channelData: redactedChannelData,
+      }
+    );
+
+    await this.sendWebhookMessage.execute({
+      eventType: WebhookEventEnum.MESSAGE_SENT,
+      objectType: WebhookObjectTypeEnum.MESSAGE,
+      payload: {
+        object: messageWebhookMapper(message, command.subscriberId, {
+          providerResponseId: result.id,
+          // for backwards compatibility
+          webhookUrl: channelData.type === ADDRESS_TYPES.WEBHOOK ? channelData.address.url : undefined,
+          channelData: redactedChannelData,
+        }),
+      },
+      organizationId: command.organizationId,
+      environmentId: command.environmentId,
+    });
+
+    return {
+      status: SendMessageStatus.SUCCESS,
+    };
+  }
+
+  private async handleMessageSendError(
+    error: unknown,
+    message: MessageEntity,
+    command: SendMessageChannelCommand,
+    channelData: ChannelData
+  ): Promise<SendMessageResult> {
+    const redactedChannelData = this.redactChannelData(channelData);
+
+    await this.sendErrorStatus(
+      message,
+      'error',
+      'unexpected_chat_error',
+      this.getErrorMessage(error) || 'Un-expect CHAT provider error',
+      command,
+      error
+    );
```

Comment:
If one message from the fanout fails, it sets this error on the entire message entity in the DB which I'm not sure if its correct, but it worked like this for the push device tokens as well so I've kept it as is for now.